aten/src/ATen/templates/UfuncCPU.cpp

```diff
@@ -0,0 +1,17 @@
+#include <ATen/native/DispatchStub.h>
+#include <ATen/TensorIterator.h>
+#include <ATen/TensorMeta.h>
+
+namespace at {
+
+// NB: this is explicitly copied here (via codegen) rather than
+// included via NativeFunctions.h to avoid recompiling this file when
+// NativeFunctions.h changes
+namespace meta {
+${meta_declaration}
```

Comment:
Peter Bell did a bunch of work to move those declarations into per-op codegen'd headers. So I think an alternative to redeclaring them here would be to codegen the includes for the relevant ops at the top of this file (`build/aten/src/ATen/ops/add_meta.h` is an example).

Reply:
Makes sense!

Reply:
Yep, this would work (though we'd still need the ifdef to do the non-per-op version in fbcode)